client/src/components/SplitList.jsx

```diff
@@ -1,100 +1,133 @@
+          const tax = totals[guestID] * (taxPercentage / 100);
+          totals[guestID] = (totals[guestID] + tip + tax).toFixed(2);
+        }
+
+        return totals;
+      };
+
+      const getEvenTotals = (guests) => {
+        const totals = {};
+        const guestArray = guests;
+        const guestIDs = [];
+        for (let i = 0; i < guestArray.length; i++) {
+          const currentGuestID = guestArray[i].guestName;
+          // const currentGuestID = guestArray[i].guest_id;
+          guestIDs.push(currentGuestID);
+          const evenTotal = (billWithTipAndTax / numberOfGuests).toFixed(2);
+          totals[currentGuestID] = evenTotal;
+        };
+        let splitEvenTotal = 0;
+        for (const guestID in totals) {
+          splitEvenTotal += Number(totals[guestID]);
+        }
+        let remainder = Math.floor((billWithTipAndTax - splitEvenTotal) * 100) / 100;
+        if (remainder > 0) {
+          while (remainder > 0) {
+            const thisGuest = guestIDs.splice(Math.floor(Math.random() * guestIDs.length), 1)[0];
+            totals[thisGuest] = (Number(totals[thisGuest]) + 0.01).toFixed(2);
+            remainder -= 0.01;
+          }
+        } else if (remainder < 0) {
+          while (remainder < 0) {
+            const thisGuest = guestIDs.splice(Math.floor(Math.random() * guestIDs.length), 1)[0];
+            totals[thisGuest] = (Number(totals[thisGuest]) - 0.01).toFixed(2);
+            remainder += 0.01;
+          }
+        }
+        return totals;
+      };
+
+      if (props.split === 'by Item') {
+        paymentsOwed = getGuestTotals(props.guests);
+      } else {
+        paymentsOwed = getEvenTotals(props.guests);
+      };
 
-  if (props.split === 'by Item') {
-    paymentsOwed = getGuestTotals(props.guests);
-  } else {
-    paymentsOwed = getEvenTotals(props.guests);
-  }
+      props.setTopLevelState(
+        'finalTotals', {
+          paymentsOwed,
+          preliminaryTotal: billTotalWithoutTipOrTax.toFixed(2),
+          tipAmount: tipAmount.toFixed(2),
+          tax: taxAmount.toFixed(2),
+          finalTotal: billWithTipAndTax.toFixed(2)
+        }
+      );
+    }).catch(function (error) {
+      console.error('GET TAX ERROR', error);
+      return null;
+    });
+  }, [props.tipPercentage, props.split]);
 
   return (
-  <ul>
-    {props.guests.map((guest, i) =>
-      <IndividualOwes key = {i} guestName = {guest.guestName} paymentOwed = {paymentsOwed[guest.guestName]} />
-    )}
-    <div>
-      <hr />
-      <div> Preliminary Total: ${billTotalWithoutTipOrTax.toFixed(2)} </div>
-      <div> Tip Amount: ${tipAmount.toFixed(2)} </div>
-      <div> Tax: ${taxAmount.toFixed(2)} </div>
-      <div> Final Total: ${billWithTipAndTax.toFixed(2)} </div>
-      <hr />
-    </div>
-  </ul>
+    <ul>
+      {props.guests.map((guest, i) =>
+        <IndividualOwes key = {i} guestName = {guest.guestName} paymentOwed = {props.finalTotals.paymentsOwed[guest.guestName] || 'Loading...'} />
```

Comment:
Kind of a nit, but I think putting the 'Loading...' into the component itself makes the concerns a little more centralized. So instead of the OR condition in line here, you can check for the truthiness in the `IndividualOwes` component instead.